source/common/http/conn_manager_impl.cc

```diff
@@ -987,6 +987,20 @@ void ConnectionManagerImpl::ActiveStreamDecoderFilter::encodeTrailers(HeaderMapP
   parent_.encodeTrailers(nullptr, *parent_.response_trailers_);
 }
 
+void ConnectionManagerImpl::ActiveStreamDecoderFilter::
+    onDecoderFilterAboveWriteBufferHighWatermark() {
+  ENVOY_STREAM_LOG(debug, "Read-disabling downstream stream due to filter callbacks.", parent_);
+  parent_.response_encoder_->getStream().readDisable(true);
+  parent_.connection_manager_.stats_.named_.downstream_flow_control_paused_reading_total_.inc();
```

Comment:
@mattklein123 Do you think it's also worth adding the source-cluster of the flow control backup or is that overkill?